### PR TITLE
Fixes some more trauma bugs

### LINKED ISSUE #35180: Phobia trauma can be triggered by names
If someone who has a word that triggers your phobia in their name says something unrelated, you will get triggered

### LINKED ISSUE #35537: CURATOR CAN STILL GET SCARED OF HEARING SNAKE WHEN HES DEAF
image

DESPITE HAVING MY EARS REMOVED I STILL HEAR SNAKE AND GET SCARED OF IT

### LINKED ISSUE #36454: Imaginary friend special trauma doesn't work
Issue reported from  Round ID: 85035 (/tg/Station Bagil [ENGLISH] [US-EAST] [100% FREE LAG])

Reporting client version: 512

If you get the trauma, I think it prompts ghosts to be X's imaginary friend, but nobody is picked even if they choose yes, and the trauma is removed. Which happened this round.

### PR DESCRIPTION
**Pull request by XDTM**
Fixes random traumas not acquiring the owner and specific traumas not applying at all. Also fixes the bug where phobias parsed the whole message instead of only the spoken line.

Fixes #35537
Fixes #35180

Might fix #36454 but i'm not sure if it's related.

### CODE REVIEW

File: code/datums/brain_damage/brain_trauma.dm [hunk 1]
```diff
@@ -12,6 +12,9 @@
 	var/can_gain = TRUE //can this be gained through random traumas?
 	var/resilience = TRAUMA_RESILIENCE_BASIC //how hard is this to cure?
 
+/datum/brain_trauma/New()
+	return
```
Comment (vuonojenmustaturska) [inline code/datums/brain_damage/brain_trauma.dm]: ???

Comment (XDTM) [reply code/datums/brain_damage/brain_trauma.dm]: oops i left it there while i was trying to fix a byond bug

File: code/modules/mob/living/brain/brain_item.dm [hunk 2]
```diff
@@ -169,6 +169,8 @@
 	if(brainmob)
 		qdel(brainmob)
 		brainmob = null
+	for(var/T in traumas)
+		qdel(T)
```
Comment (Cyberboss) [inline code/modules/mob/living/brain/brain_item.dm]: QDEL_LIST

File: code/modules/mob/living/brain/brain_item.dm [hunk 3]
```diff
@@ -169,6 +169,8 @@
 	if(brainmob)
 		qdel(brainmob)
 		brainmob = null
```
Comment (Cyberboss) [inline code/modules/mob/living/brain/brain_item.dm]: QDEL_NULL

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Pull request merged**